### ISSUE TITLE
        entropy_loss = (mac_out_entropy * entropy_mask).sum() / entropy_mask.sum()

### ISSUE THREAD

**Issue by xu-dong-yuan** [mzho7212/LICA#8]
Could you tell me what this line of code means, or what the mathematical formula it represents?
thank you

**Comment by mzho7212** [comment]
Hi

As one batch of the training data includes multiple rounds of the game and each round of the game has a different number of total steps, (one round of game finishes when all agent controlled units are dead or all enemy units are dead) each round's data is padded into the same dimension for model training purpose. 

And this line is basically masking out the padded data's model output as they should not contribute to the training. 

Hope this helps.

**Comment by xu-dong-yuan** [comment]
Is this the code to implement adaptive entropy？
The result of entropy_ratio * entropy_loss  is entropy_coef.
I don't known how to adjust the entropy?

**Comment by mzho7212** [comment]
The adaptive entropy implementation is in https://github.com/mzho7212/LICA/blob/main/src/learners/lica_learner.py Line 74 - Line 77. And the specific line you quoted (Line 74) is for computing the regular entropy term. 

The adaptive entropy term is "entropy_ratio * entropy_loss" where the entropy ratio is adaptively computed based on the stochasticity of the policy. In our implementation, we choose the policy entropy value itself to measure the policy stochasticity. (Note that there are also other ways to measure policy stochasticity as we discussed in Section 3.3 of our paper) The numeric value of entropy_ratio * entropy_loss is indeed entropy_coef, whereas the underlying magnitude of the derivative is adjusted adaptively. 

You can think of entropy_ratio as a new coef constant applied to the entropy loss term, and you can still adjust the strength of entropy regularization by adjusting the entropy_coef in the config file.

**Comment by xu-dong-yuan** [comment]
Thank you for your reply
